templates/NOTES.txt

```diff
@@ -1,26 +1,107 @@
+  Target Memory at {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}%
+{{- else }}
+  [auto-scaling is disabled]
 {{- end }}
+
+Node Selector:
+  {{ default "[no node selectors applied]" .Values.nodeSelector }}
+
+Node Affinity:
+  {{ default "[no node affinity applied]" .Values.affinity }}
+
+Node Tolerations:
+{{- if .Values.tolerations }}
+  {{- range $toleration := .Values.tolerations }}
+  Key: {{ $toleration.key }} 
+  Operator: {{ $toleration.operator }} 
+  Effect: {{ $toleration.effect }}
+  {{- end }}
+{{- else }}
+  [no node tolarations applied]
 {{- end }}
+
+Pod Annotations Applied:
+  {{ default "[no annotations applied]" .Values.podAnnotations }}
+
+Pod Security Context:
+  {{ default "[no pod security context applied]" .Values.podSecurityContext }}
+
+Container Security Context:
+  {{ default "[no container security context applied]" .Values.securityContext }}
+
+Network Service: 
+  Type: {{ .Values.service.type }}
+  Ports: 
+    {{- range $port := .Values.service.ports }}
+    {{ $port.name }}: {{ $port.targetPort }} [internal] <== {{ $port.port }} [external]
+    {{- end }}
+
+Network Ingress:
+  {{- if .Values.ingress.enabled }}
+  Application URLs:
+    {{- range $host := .Values.ingress.hosts }}
+      {{- range .paths }}
+    http{{ if $.Values.ingress.tls }}s{{ end }}://{{ $host.host }}{{ .path }}
+      {{- end }}
+    {{- end }}
+  Ingress Annotations:  
+    {{ default "[no ingress annotations applied]" .Values.ingress.annotations }}
+  {{- else }}
+    Run the following in a *nix terminal to access the running
+    {{- if contains "NodePort" .Values.service.type }}
+      {{- " NodePort service endpoint" }}
+      export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "optimizely-agent.fullname" . }})
+      export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+      echo http://$NODE_IP:$NODE_PORT
+    {{- else if contains "LoadBalancer" .Values.service.type }}
+      {{- " LoadBalancer service endpoint" }}
+        NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+              You can watch the status of by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "optimizely-agent.fullname" . }}'
+      export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "optimizely-agent.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+      echo http://$SERVICE_IP:{{ .Values.service.port }}
+    {{- else if contains "ClusterIP" .Values.service.type }}
+      {{- " ClusterIP service IP address" }}
+      export CLUSTER_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "optimizely-agent.fullname" . }} -o jsonpath='{.spec.clusterIP}')
+      echo http://$CLUSTER_IP:8080
+    {{- end }}
```

Comment:
@msohailhussain I thought it was good to keep the outputs from the other service types. 😄 

My research points to our need to run `kubectl` commands to access info from the running deployments like IP, Ports, etc.  It appears that helm resolves templates earlier before the deployment is necessarily online ie no access to post-install values in the NOTES. ...again, from what I can tell.

I'm certain @yasirfolio3 can take this further than my knowledge/experience.